CMake 4.1.0+ fails to configure cross-build for macOS with Ninja generator

When cross-compiling for macOS on Linux, the configuration step fails when using the Ninja generator:
```
$ gmake -j $(nproc) -C depends HOST=x86_64-apple-darwin NO_QT=1
$ cmake -B build -G Ninja --toolchain depends/x86_64-apple-darwin/toolchain.cmake
<snip>
CMake Error at src/CMakeLists.txt:301 (add_executable):
  The install of the bitcoind target requires changing an RPATH from the
  build tree, but this is not supported with the Ninja generator unless on an
  ELF-based or XCOFF-based platform.  The CMAKE_BUILD_WITH_INSTALL_RPATH
  variable may be set to avoid this relinking step.


<snip>
```

This behavior is observed with CMake starting from version 4.1.0.

This issue does not occur when building with depends on macOS.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,8 +69,13 @@ set(CLIENT_BUGREPORT "https://github.com/bitcoin/bitcoin/issues")
 #=============================
 # Language setup
 #=============================
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
+if(CMAKE_VERSION VERSION_LESS 4.2 AND CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
   # We do not use the install_name_tool when cross-compiling for macOS.
+  # However, CMake < 4.2 still searches for Apple's version of the tool,
+  # which causes an error during configuration.
+  # See:
+  # - https://gitlab.kitware.com/cmake/cmake/-/issues/27069
+  # - https://gitlab.kitware.com/cmake/cmake/-/merge_requests/10955
   # So disable this tool check in further enable_language() commands.
   set(CMAKE_PLATFORM_HAS_INSTALLNAME FALSE)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,7 +77,7 @@ if(CMAKE_VERSION VERSION_LESS 4.2 AND CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NO
   # - https://gitlab.kitware.com/cmake/cmake/-/issues/27069
   # - https://gitlab.kitware.com/cmake/cmake/-/merge_requests/10955
   # So disable this tool check in further enable_language() commands.
-  set(CMAKE_PLATFORM_HAS_INSTALLNAME FALSE)
+  set(CMAKE_INSTALL_NAME_TOOL "${CMAKE_COMMAND} -E true")
 endif()
 enable_language(CXX)
 set(CMAKE_CXX_STANDARD 20)

